lib/src/binary_codec.dart

```diff
@@ -467,8 +476,22 @@ class PostgresBinaryDecoder<T> extends Converter<Uint8List?, T?> {
         return buffer.getFloat64(0) as T;
       case PostgreSQLDataType.timestampWithoutTimezone:
       case PostgreSQLDataType.timestampWithTimezone:
-        return DateTime.utc(2000)
-            .add(Duration(microseconds: buffer.getInt64(0))) as T;
+        try {
+          final value = buffer.getInt64(0);
+          // SELECT * FROM pg_authid where rolname ='sw.mqueiroz'
```

Comment:
Please add more context to the values here.

Reply:
I detected a problem with this driver dealing with the date/timestamp of PostgreSQL 8.2.21, it may be that this problem occurs with other versions of postgresql as well, the problem is that it interprets the date incorrectly, instead of returning the value 725968771477000 to the date 2023-01-02 09:59:31.477Z it returns 4739373075810553430 which is 152184-10-02 01:03:30.553430Z

I tested with some versions of postgresql:
```
          // value = 4739373075810553430 PostgreSQL 8.2.21 bug
          // value =  725968771477000 PostgreSQL PostgreSQL 15.3
          // value = 725968771477000 PostgreSQL 14.6
          // value = 725968771477000 PostgreSQL 8.1.11
          // value = 725968771477000 PostgreSQL 8.2.23
```